plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/FBTImporter.java

```diff
@@ -449,6 +486,12 @@ private boolean handleBaseFBChildren(final BaseFBType type, final String name)
 						action.setAlgorithm(alg);
 					}
 				}
+				final List<SimpleECAction> simplelist = algorithmNameSimpleECActionMapping.get(alg.getName());
+				if (null != simplelist) {
+					for (final SimpleECAction action : simplelist) {
+						action.setAlgorithm(alg);
+					}
+				}
```

Comment:
Please note that the current implementation will silently skip over any non-existent algorithms. This may easily happen, for example, when saving with syntax errors in the ST code. It is also currently impossible for users to see or repair it in the UI.

I believe we need to either create error marker algorithms (and also manage them in the ST reconciler) or just refer from actions to algorithms by name (and add a corresponding check in the ST validator).

Reply:
Haven't worked much with either error markers or the validators. What would be the better way to handle this with consideration to a future editor for this?

Reply:
I believe it would definitely be more robust to refer from actions to algorithms by name, particularly regarding the ST editor synchronization and reconciler.

A future editor would probably use a drop down field based on the names of the currently available algorithms in either case. If we use name-based references, we could potentially even allow the user to input arbitrary values in the future.

Reply:
Simple FBs could also be a nice starting point for testing this out as starting point of later doing this in BasicFBs.